service/controller/resource/capzcrs/create_test.go

```diff
@@ -114,6 +125,76 @@ func Test_AzureConfigCRMapping(t *testing.T) {
 	}
 }
 
+func Test_mergeAzureCluster(t *testing.T) {
+	testCases := []struct {
+		name         string
+		original     *capzv1alpha3.AzureCluster
+		desired      *capzv1alpha3.AzureCluster
+		expected     *capzv1alpha3.AzureCluster
+		errorMatcher func(error) bool
+	}{
+		{
+			name:         "case 0: empty values",
+			original:     &capzv1alpha3.AzureCluster{},
+			desired:      &capzv1alpha3.AzureCluster{},
+			expected:     &capzv1alpha3.AzureCluster{},
+			errorMatcher: nil,
+		},
+		{
+			name: "case 1: preserve subnets",
+			original: &capzv1alpha3.AzureCluster{
+				Spec: capzv1alpha3.AzureClusterSpec{
+					NetworkSpec: capzv1alpha3.NetworkSpec{
+						Subnets: []*capzv1alpha3.SubnetSpec{
+							&capzv1alpha3.SubnetSpec{
+								ID:   "nodepool-np201",
+								Role: capzv1alpha3.SubnetNode,
+							},
+						},
+					},
+				},
+			},
+			desired: &capzv1alpha3.AzureCluster{},
```

Comment:
Would it be useful to have a test case where there are, let's say, 2 desired and 3 original? An example where there is a diff, but both are non-null?

Reply:
Mapping from `AzureConfig` -> `AzureCluster` doesn't recognize subnets at all. Therefore desired would always have nil for subnets. So: No, it wouldn't be useful :slightly_smiling_face:

Reply:
Thanks!